impeller/renderer/backend/vulkan/capabilities_vk.cc

```diff
@@ -348,6 +348,21 @@ bool CapabilitiesVK::SetPhysicalDevice(const vk::PhysicalDevice& device) {
              .supportedOperations &
          vk::SubgroupFeatureFlagBits::eArithmetic);
 
+  {
+    // Query texture support.
+    // TODO(jonahwilliams):
+    // https://github.com/flutter/flutter/issues/129784
+    vk::PhysicalDeviceMemoryProperties memory_properties;
+    device.getMemoryProperties(&memory_properties);
+
+    for (auto i = 0u; i < memory_properties.memoryTypeCount; i++) {
+      if (memory_properties.memoryTypes[i].propertyFlags &
+          vk::MemoryPropertyFlagBits::eLazilyAllocated) {
+        supports_memoryless_textures_ = true;
```

Comment:
I sound like a broken record, but I don't think one can make the leap.  I'm reading through https://registry.khronos.org/vulkan/specs/1.3-extensions/html/vkspec.html#memory-device-lazy_allocation and it doesn't talk about this.

Maybe there is an extension that can give us an exact query for this support?

Reply:
I understand that you are not convinced this is sufficient. However, lacking evidence of a phone where this is an issue I  think we should focus on the existing pile of concrete problems.

Reply:
Yea, fair point.  The Android ecosystem is so diverse, it's difficult to know if there is a device that would pass this test but have better performance caching the textures instead.  As a tools developer too we don't have analytics or anything that will tip us off if there is a problem.  I think it's worth at least looking, I was hoping there some documentation somewhere about this.

Reply:
Here is the documentation I could find.  The name vulkan gives this is "Transient Attachments": https://github.com/KhronosGroup/Vulkan-Samples/tree/main/samples/performance/subpasses#transient-attachments

We should use that nomenclature instead of memoryless textures.  I couldn't find a more apropos capabilities check than this.

Reply:
memoryless is consistent with the nomenclature of the Metal backend and is what @chinmaygarde asked me to use here.